lib/seahorn/BvOpSem2Allocators.cc

```diff
@@ -296,7 +298,23 @@ class StaticOpSemAllocator : public OpSemAllocator {
       preAlloc(inst, memSz, true);
     } else {
       // -- allocate max allowed for dynamically sized allocations
-      preAlloc(inst, m_maxSymbAllocSz, false);
+      // -- use inferred upper bound if crab analysis is performed
+      auto operand_inst = dyn_cast<llvm::Instruction>(inst.getOperand(0));
+      auto rangeValCrab = m_sem.getCrabInstRng(*operand_inst);
+      auto rangeValLVI = m_sem.getLVIInstRng(*operand_inst);
+      uint64_t ub_crab = rangeValCrab.getUpper().getLimitedValue();
+      uint64_t ub_lvi = rangeValLVI.getUpper().getLimitedValue();
+      LOG("opsem-rng", MSG << "Inferred upper bound for " << *operand_inst
+                           << ", Crab: " << ub_crab << ", LVI: " << ub_lvi
+                           << "\n";);
+      if (rangeValCrab.isFullSet() || ub_crab >= UINT_MAX) {
```

Comment:
What is the purpose of having `ub_crab >= UINT_MAX` ?

Reply:
The `preAlloc` method requires byte as an `unsigned` type, but our inferred range has type `uint64_t`. Maybe change it to unsigned?